docs/temporal.md

```diff
@@ -0,0 +1,229 @@
+)
+
+temporal_agent = TemporalAgent(agent)  # (1)!
+
+
+@workflow.defn
+class GeographyWorkflow:  # (2)!
+    @workflow.run
+    async def run(self, prompt: str) -> str:
+        result = await temporal_agent.run(prompt)  # (3)!
+        return result.output
+
+
+async def main():
+    client = await Client.connect(  # (4)!
+        'localhost:7233',  # (5)!
+        plugins=[PydanticAIPlugin()],  # (6)!
+    )
+
+    async with Worker(  # (7)!
+        client,
+        task_queue='geography',
+        workflows=[GeographyWorkflow],
+        plugins=[AgentPlugin(temporal_agent)],  # (8)!
+    ):
+        output = await client.execute_workflow(  # (9)!
+            GeographyWorkflow.run,
+            args=['What is the capital of Mexico?'],
+            id=f'geography-{uuid.uuid4()}',
+            task_queue='geography',
+        )
+        print(output)
+        #> Mexico City (Ciudad de México, CDMX)
+```
+
+1. The original `Agent` cannot be used inside a deterministic Temporal workflow, but the `TemporalAgent` can.
+2. As explained above, the workflow represents a deterministic piece of code that can use non-deterministic activities for operations that require IO.
+3. [`TemporalAgent.run()`][pydantic_ai.durable_exec.temporal.TemporalAgent.run] works just like [`Agent.run()`][pydantic_ai.Agent.run], but it will automatically offload model requests, tool calls, and MCP server communication to Temporal activities.
+4. We connect to the Temporal server which keeps track of workflow and activity execution.
+5. This assumes the Temporal server is [running locally](https://github.com/temporalio/temporal#download-and-start-temporal-server-locally).
+6. The [`PydanticAIPlugin`][pydantic_ai.durable_exec.temporal.PydanticAIPlugin] tells Temporal to use Pydantic for serialization and deserialization, and to treat [`UserError`][pydantic_ai.exceptions.UserError] exceptions as non-retryable.
+7. We start the worker process that will listen on the specified task queue and run workflows and activities.
+8. The [`AgentPlugin`][pydantic_ai.durable_exec.temporal.AgentPlugin] registers the `TemporalAgent`'s activities with the worker.
+9. We call on the server to execute the workflow on a worker that's listening on the specified task queue.
+10. The agent's `name` is used to uniquely identify its activities.
+
+_(This example is complete, it can be run "as is" — you'll need to add `asyncio.run(main())` to run `main`)_
+
+In a real world application, the agent, workflow, and worker are typically defined separately from the code that calls for a workflow to be executed.
+Because Temporal workflows need to be defined at the top level of the file and the `TemporalAgent` instance is needed inside the workflow and when starting the worker (to register the activities), it needs to be defined at the top level of the file as well.
+
+For more information on how to use Temporal in Python applications, see their [Python SDK guide](https://docs.temporal.io/develop/python).
```

Comment:
```suggestion
Note that temporal workflows are useful for more than just handling failures, including .... For more information on how to use Temporal in Python applications, see their [Python SDK guide](https://docs.temporal.io/develop/python).
```
(the "..." here might talk about how you can use signals to achieve human-in-the-loop, how you can "sleep" for days without a problem, etc., I just think it's worth pointing out that temporal is more than just error handling, and helps motivate why someone might want to read more. Sorry I didn't flesh it out for you, maybe AI can help lol.)